### PR TITLE
Fixed #29573 -- Added link from per-item aggregation topic discussion to `annotate()` ref.

### PR DESCRIPTION
**Pull request by VishvajitP**
Fixed [29573](https://code.djangoproject.com/ticket/29573)

Links for `QuerySet`, `annotate`,  `aggregate` are added.

### CODE REVIEW

File: docs/topics/db/aggregation.txt [hunk 1]
```diff
@@ -99,19 +99,20 @@ In a hurry? Here's how to do common aggregate queries, assuming the models above
     >>> pubs[0].num_books
     1323
 
-Generating aggregates over a ``QuerySet``
-=========================================
+Generating aggregates over a QuerySet
+=====================================
```
Comment (carltongibson) [inline docs/topics/db/aggregation.txt]: `QuerySet` is a code item, so we probably do want the backticks.

File: docs/topics/db/aggregation.txt [hunk 2]
```diff
@@ -138,28 +140,32 @@ by providing that name when you specify the aggregate clause::
...
 the maximum and minimum price of all books, we would issue the query::
 
     >>> from django.db.models import Avg, Max, Min
     >>> Book.objects.aggregate(Avg('price'), Max('price'), Min('price'))
     {'price__avg': 34.35, 'price__max': Decimal('81.20'), 'price__min': Decimal('12.99')}
 
-Generating aggregates for each item in a ``QuerySet``
-=====================================================
+Generating aggregates for each item in a :class:`~django.db.models.query.QuerySet`
+==================================================================================
```
Comment (carltongibson) [inline docs/topics/db/aggregation.txt]: So I think just use backticks in titles, with cross references in the body text.

File: docs/topics/db/aggregation.txt [hunk 3]
```diff
@@ -309,25 +315,25 @@ file::
 (The resulting dictionary will have a key called ``'average_rating'``. If no
 such alias were specified, it would be the rather long ``'book__rating__avg'``.)
 
-Aggregations and other ``QuerySet`` clauses
-===========================================
+Aggregations and other QuerySet clauses
+=======================================
```
Comment (carltongibson) [inline docs/topics/db/aggregation.txt]: Again, please retain the backticks.

**Comment by VishvajitP** [comment]
@carltongibson Thanks for the review. Fixes have been added.

**Comment by timgraham** [comment]
I'm not sure if adding links everywhere is of much value. Doc translators may need to redo all the sentences. Also, instead of 
```
:meth:`~django.db.models.query.QuerySet.aggregate`
```
I think it would be fine to use the shorter forms like:
```
:meth:`~.QuerySet.aggregate`
```

**Comment by carltongibson** [comment]
Shorter forms would be great. The file already has the `currentmodule:` declaration:

```
.. currentmodule:: django.db.models
```

> I'm not sure if adding links everywhere is of much value...

Perhaps it would be enough/better to just link (for example) the first occurrence in the [generating-aggregates-for-each-item-in-a-queryset](https://docs.djangoproject.com/en/2.1/topics/db/aggregation/#generating-aggregates-for-each-item-in-a-queryset) section? (That was the initial thought from the discussion on the ticket.)

**Comment by VishvajitP** [comment]
I thought it'd be useful for the end user to have the references at all the places. But after the discussion here,  I have reverted the code to link just the mentioned places in the ticket.

**Comment by VishvajitP** [comment]
Thanks @carltongibson. I am facing issues while squashing the commits. 
I am getting `error: cannot 'squash' without a previous commit` persistently. Could you please help me on this ?

**Comment by VishvajitP** [comment]
@carltongibson Thanks for the help. Could you also help me with the steps you follow to squash these commits ?

**Comment by carltongibson** [comment]
Hi @VishvajitP. Sure: 

1. I [`git reset`](https://git-scm.com/docs/git-reset) to ccf870ebf5a9f9b44711a08549833f97ef3f7a41, which was the base commit for your branch. This leaves your changes in place but uncommitted. 
2. I committed the chunks for the first commit, specifying you as the author. 
3. Same for the cosmetic changes. 
4. Rebases on `origin/master`.
5. Forced pushed to this branch to replace the history. 

At step 1, you could `rebase -i`, picking your first commit and squashing all the others, but then you still need to separate the cosmetic changes, hence I went for `reset` instead.

**Comment by timgraham** [comment]
I removed the line wrapping commit as that's not something important enough to complicate commit history with.

**Pull request merged**